include/forcing/Forcing.h

```diff
@@ -228,20 +242,32 @@ class Forcing
                 //Row vector
                 std::vector<std::string>& vec = data_list[i];
 
-                //Declare struct for the current row date-time
-                struct tm current_row_date_time;
+                //Declare struct for the current row date-time 
+                struct tm current_row_date_time_utc;
 
                 //Allocate memory to struct for the current row date-time
-                current_row_date_time = tm();
+                current_row_date_time_utc = tm();
 
                 //Grab time string from first column
                 string time_str = vec[0];
 
                 //Convert time string to time struct
-                strptime(time_str.c_str(), "%Y-%m-%d %H:%M:%S", &current_row_date_time);
+                strptime(time_str.c_str(), "%Y-%m-%d %H:%M:%S", &current_row_date_time_utc);
 
-                //Convert current row date-time to epoch time
-                time_t current_row_date_time_epoch = mktime(&current_row_date_time);
+                //Convert current row date-time UTC to epoch time
+                time_t current_row_date_time_epoch = timegm(&current_row_date_time_utc);
+
+                //Ensure that forcing data covers the entire model period. Otherwise, throw an error.
+                if (i == 1 && start_date_time_epoch < current_row_date_time_epoch)
+                    /// \todo TODO: Return appropriate error
+                    //cout << "WARNING: Forcing data begins after the model start time." << endl;
+                    throw std::runtime_error("Error: Forcing data begins after the model start time.");
+
+
+                else if (i == data_list.size() - 1 && current_row_date_time_epoch < end_date_time_epoch)
+                    /// \todo TODO: Return appropriate error
+                    cout << "WARNING: Forcing data ends before the model end time." << endl;
```

Comment:
Warning or error for this?

Reply:
A warning for now, and proper semantics/runtime warning when we run out of forcing.

Reply:
Okay. I have that in the get next function.